src/poll.rs

```diff
@@ -699,8 +699,24 @@ mod notify {
 
         /// Pops a notification (if any) from the pipe.
         pub(super) fn pop_notification(&self) -> Result<(), io::Error> {
+            // Pipes on Vita do not guarantee that after `write` call succeeds, the
+            // data becomes immediately available for reading on the other side of the pipe.
+            // To ensure that the notification is not lost, the read side of the pipe is temporarily
+            // switched to blocking for a single `read` call.
+            #[cfg(target_os = "vita")]
+            rustix::fs::fcntl_setfl(
+                &self.read_pipe,
+                rustix::fs::fcntl_getfl(&self.read_pipe)? & !rustix::fs::OFlags::NONBLOCK,
+            )?;
+
             read(&self.read_pipe, &mut [0; 1])?;
```

Comment:
I'm worried that, if `read` errors out, the system call setting the pipe back to non-blocking won't be called. Instead of the `?` here, grab the result to make sure that the non-blocking syscall is still called.

Reply:
Makes sense, done